sky/client/sdk_async.py

```diff
@@ -0,0 +1,541 @@
+@server_common.check_server_healthy_or_start
+@annotations.client_api
+async def get(request_id: str) -> Any:
+    """Async version of get() that waits for and gets the result of a request.
+
+    Args:
+        request_id: The request ID of the request to get.
+
+    Returns:
+        The ``Request Returns`` of the specified request. See the documentation
+        of the specific requests above for more details.
+
+    Raises:
+        Exception: It raises the same exceptions as the specific requests,
+            see ``Request Raises`` in the documentation of the specific requests
+            above.
+    """
+    async with aiohttp.ClientSession() as session:
+        async with session.get(
+                f'{server_common.get_server_url()}'
+                f'/api/get?request_id={request_id}',
+                timeout=aiohttp.ClientTimeout(
+                    total=None,
+                    connect=client_common.
+                    API_SERVER_REQUEST_CONNECTION_TIMEOUT_SECONDS),
+                cookies=server_common.get_api_cookie_jar()) as response:
+            request_task = None
+            if response.status == 200:
+                request_task = requests_lib.Request.decode(
+                    requests_lib.RequestPayload(**await response.json()))
+            elif response.status == 500:
+                try:
+                    request_task = requests_lib.Request.decode(
+                        requests_lib.RequestPayload(**await response.json()))
+                    logger.debug(f'Got request with error: {request_task.name}')
+                except Exception:  # pylint: disable=broad-except
+                    request_task = None
+            if request_task is None:
+                with ux_utils.print_exception_no_traceback():
+                    raise RuntimeError(
+                        f'Failed to get request {request_id}: '
+                        f'{response.status} {await response.text()}')
+            error = request_task.get_error()
+            if error is not None:
+                error_obj = error['object']
+                if env_options.Options.SHOW_DEBUG_INFO.get():
+                    stacktrace = getattr(error_obj, 'stacktrace',
+                                         str(error_obj))
+                    logger.error('=== Traceback on SkyPilot API Server ===\n'
+                                 f'{stacktrace}')
+                with ux_utils.print_exception_no_traceback():
+                    raise error_obj
+            if request_task.status == requests_lib.RequestStatus.CANCELLED:
+                with ux_utils.print_exception_no_traceback():
+                    raise exceptions.RequestCancelled(
+                        f'{colorama.Fore.YELLOW}Current {request_task.name!r} '
+                        f'request ({request_task.request_id}) is cancelled by '
+                        f'another process. {colorama.Style.RESET_ALL}')
+            return request_task.get_return_value()
```

Comment:
This error handling is mostly the same sdk.get() (sync version). Can we split it into a separate common func?

Reply:
i really really dislike how some python library developers always make their interfaces only slightly different. `aiohttp.ClientResponse` has `status` and `requests.Response` has `status_code`. And with the `aiohttp` one you need to do `await` for `text` and `json`. I can extract this into another function that takes the `json` and `status_code` as parameters.